services/harmony/test/destination-url.ts

```diff
@@ -72,15 +72,15 @@ describe('when setting destinationUrl on ogc request', function () {
     it('the job has harmony-job-status-link file created with the job status link', async function () {
       expect(this.res.status).to.equal(200);
       const jobId = JSON.parse(this.res.text).jobID;
-      const s3Url = 's3://dummy/p1/' + jobId + '/harmony-job-status-link';
+      const s3Url = 's3://dummy/P1/' + jobId + '/harmony-job-status-link';
       const statusLink = await defaultObjectStore().getObject(s3Url);
       // this.res.request.url is the job status link
       expect(statusLink).to.equal(this.res.request.url);
     });
 
     it('sets the destination_url on the job in db', async function () {
       const retrieved = await Job.forUser(this.trx, 'anonymous');
-      expect(retrieved.data[0].destination_url).to.eq('s3://dummy/p1');
+      expect(retrieved.data[0].destination_url).to.eq('s3://dummy/P1');
```

Comment:
nit: I think when you are fixing a bug like this it makes more sense to add a separate test just for that bug, rather than co-opting an existing test. In the future someone might change this URL to be all lower-case for some other reason, because it's not obvious that this is also verifying that case is preserved.

Reply:
Our test is based on File store, not real s3 store. So this test will not test the real s3 scenario that is in this ticket. I changed the capitalization just indicate the the s3 key can be mixed case.

Reply:
I thought the bug that Frank reported was just what you fixed - that the whole URL was being converted to lower-case when it should not be (just the s3:// part). In which case you are testing that by using a mixed-case URL. I was just saying that it's not obvious that this is being verified by this test.

Reply:
It will not be verified by the test because we use file store for testing which is case insensitive. I don't think it is worth it to change the testing mock or add a unit test.

Reply:
@indiejames Great observation that the test will fail in github as it runs on linux. I have added a separate test as you suggested. Thanks!